File: setMismatch.py
from typing import List

class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n = len(nums)
        num_set = set()
        duplicate = -1

        for num in nums:
            if num in num_set:
                duplicate = num
            num_set.add(num)

        for i in range(1, n + 1):
            if i not in num_set:
                return [duplicate, i]

File: setMismatch.go
func findErrorNums(nums []int) []int {
    n := len(nums)
    numMap := make(map[int]bool)
    duplicate := -1

    for _, num := range nums {
        if numMap[num] {
            duplicate = num
        }
        numMap[num] = true
    }

    for i := 1; i <= n; i++ {
        if !numMap[i] {
            return []int{duplicate, i}
        }
    }

    return nil
}
